website/docs/concepts/wasm-bindgen/web-sys.md

```diff
@@ -0,0 +1,174 @@
+[`web_sys::Node`](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Node.html). The
+`NodeRef::get` method will return a `Option<Node>` value, however, most of the time in Yew you want
+to cast this value to a specific element so you can use it's specific methods. This casting
+can be done using [`JsCast`](../wasm-bindgen#JsCast) on the `Node` value, if present, but Yew
+provides the `NodeRef::cast` method to perform this casting for convenience and so that you don't
+necessarily have to include the `wasm-bindgen` dependency for the `JsCast` trait.
+
+The two code blocks below do essentially the same thing, the first is using `NodeRef::cast` and
+the second is using [`JsCast::dyn_into`](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/trait.JsCast.html#method.dyn_into)
+on the `web_sys::Node` returned from `NodeRef::get`.
+
+```rust
+use yew::{web_sys::HtmlInputElement, NodeRef};
+
+fn with_node_ref_cast(node_ref: NodeRef) {
+	if let Some(input) = node_ref.cast::<HtmlInputElement>() {
+		// do something with HtmlInputElement
+	}
+}
+```
+
+```rust
+use yew::{web_sys::HtmlInputElement, NodeRef};
+use wasm_bindgen::JsCast;
+
+fn with_jscast(node_ref: NodeRef) {
+	if let Some(input) = node_ref
+		.get()
+		.and_then(|node| node.dyn_into::<HtmlInputElement>().ok()) {
+		// do something with HtmlInputElement
+	}
+}
+```
+
+## JavaScript example to Rust
+
+This section is to help show that any examples that use JavaScript to interact with the Web APIs
+can be adapted and written using Rust with `web-sys`.
+
+The JavaScript example:
+
+```js
+document.getElementById('mousemoveme').onmousemove = (e) => {
+    // e = Mouse event.
+    var rect = e.target.getBoundingClientRect();
+    var x = e.clientX - rect.left; //x position within the element.
+    var y = e.clientY - rect.top;  //y position within the element.
+    console.log("Left? : " + x + " ; Top? : " + y + ".");
+}
+```
+
+In Yew you will mostly be creating [`Callback`](../components/callbacks)s to use in the
+[`html!`](../html) macro so the example is going to use this approach instead of completely copying
+the approach above (but you could still do it!):
+
+```toml title=Cargo.toml
+[dependencies.web-sys]
+version = "0.3"
+# We need to enable the `DomRect` feature in order to use the
+# `get_bounding_client_rect` method.
+features = [
+	"DomRect"
+]
+
+```
+
+```rust
+use yew::{
+    html,
+	web_sys::{console, HtmlElement},
+    Callback, MouseEvent, TargetCast,
+};
+
+let onmousemove = Callback::from(|e: MouseEvent| {
+    if let Some(target) = e.target_dyn_into::<HtmlElement>() {
+        let rect = target.get_bounding_client_rect();
+        let x = (e.client_x() as f64) - rect.left();
+        let y = (e.client_y() as f64) - rect.top();
+        console::log_1(&format!("Left? : {} ; Top? : {}", x, y).into());
+    }
+});
+
+html! {
+    <div id="mousemoveme" {onmousemove}></div>
+};
+```
```

Comment:
I'm not sure whether adapting this for Yew is the best approach or not? 
It shows how you'd mostly likely adapt the JS example into Yew but skips out some of the
comparison to doing all that by hand with Rust.
Maybe show both? do it manually with all the `Closure` stuff and then showing this version
with Rust as a way of demonstrating how much easier Yew makes it?

Reply:
I like the idea of a comparison as it also teaches the user how it could be done manually if there's a need for more flexibility.